emu/src/cpu/psr.rs

```diff
@@ -186,34 +186,6 @@ impl From<Psr> for u32 {
     }
 }
 
-/// Represents the CPU state (ARM/THUMB).
-pub enum CpuState {
-    /// Which operates with 16-bit, halfword-aligned THUMB instructions.
-    /// In this state, the PC uses bit 1 to select between alternate halfwords.
-    Thumb,
-
-    /// Which executes 32-bit, word-aligned ARM instructions.
-    Arm,
-}
-
-impl From<CpuState> for bool {
-    fn from(state: CpuState) -> Self {
-        match state {
-            CpuState::Arm => false,
-            CpuState::Thumb => true,
-        }
-    }
-}
-
-impl From<bool> for CpuState {
-    fn from(state: bool) -> Self {
-        match state {
-            true => Self::Thumb,
-            false => Self::Arm,
-        }
-    }
-}
-
```

Comment:
I'm not sure about moving this to `cpu/cpu_modes.rs`. The CpuState is inside the PSR register, I've the feeling that it makes more sense to stay here but dunno